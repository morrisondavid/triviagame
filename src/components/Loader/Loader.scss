.loader-wrapper {
  width: 100%;
  height: 100%;
}

.loader-wrapper__loader {
  display: flex;
  flex-direction: column;
  height: 100%;
  align-items: center;
  justify-content: center;
  text-align: center;
}

.loader-wrapper__loader .sk-spinner {
  margin: 15px auto;
}
