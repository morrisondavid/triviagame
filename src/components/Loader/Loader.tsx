import React from "react";
import Spinner from "react-spinkit";
import "./Loader.scss";

// http://kyleamathews.github.io/react-spinkit/
const Loader = ({ text, type, ...rest }: ILoaderProps) => (
  <div className="loader-wrapper" {...rest}>
    <div className="loader-wrapper__loader">
      <Spinner fadeIn="none" name={type || "folding-cube"} />
      {text || "Loading..."}
    </div>
  </div>
);

export type LoaderTypes =
  | "three-bounce"
  | "double-bounce"
  | "rotating-plane"
  | "folding-cube"
  | "wave"
  | "wandering-cubes"
  | "pulse"
  | "chasing-dots"
  | "circle"
  | "cube-grid"
  | "wordpress"
  | "ball-grid-beat"
  | "ball-grid-pulse"
  | "line-spin-fade-loader"
  | "ball-spin-fade-loader"
  | "ball-pulse-rise"
  | "line-scale"
  | "line-scale-pulse-out"
  | "line-scale-pulse-out-rapid"
  | "line-scale-party"
  | "ball-triangle-path"
  | "ball-scale-ripple-multiple"
  | "ball-pulse-sync"
  | "ball-beat"
  | "ball-scale-multiple"
  | "ball-zig-zag"
  | "ball-zig-zag-deflect"
  | "ball-clip-rotate"
  | "ball-clip-rotate-pulse"
  | "ball-clip-rotate-multiple"
  | "ball-scale-ripple"
  | "triangle-skew-spin"
  | "pacman";

interface ILoaderProps {
  text?: string;
  type?: LoaderTypes;
}

export default Loader;
