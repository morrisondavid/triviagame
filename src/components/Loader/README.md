# \<Loader \/\> Component

Creates a simple loader using Spinkit's react component (https://github.com/tobiasahlin/SpinKit) `react-spinkit`.

## Usage

```javascript
import Loader from "components/Loader/Loader";

<Loader text="Loading text..." type="cube-grid" />;
```

### Props

* `text` (optional) - What text to show. Defaults to "Loading..."
* `type` (optional) - This is the type of spinkit loader to show. See http://tobiasahlin.com/spinkit/. Defaults to `cube-grid`
