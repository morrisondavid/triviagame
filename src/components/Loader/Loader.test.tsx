import React from "react";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Loader from "./Loader";

describe("<Loader />", () => {
  test("#render", () => {
    const component = renderer.create(<Loader />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("#render has <StyledLoaderWrapper />", () => {
    const component = shallow(<Loader />);
    const wrapper = component.find("StyledLoaderWrapper");

    expect(wrapper).toBeDefined();
  });

  test("#render has <StyledLoader />", () => {
    const component = shallow(<Loader />);
    const loader = component.find("StyledLoader");

    expect(loader).toBeDefined();
  });

  test("#render has <Spinner />", () => {
    const component = shallow(<Loader />);
    const spinner = component.find("Spinner");

    expect(spinner).toBeDefined();
  });
});
