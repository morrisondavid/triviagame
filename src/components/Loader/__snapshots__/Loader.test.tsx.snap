// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> #render 1`] = `
<div
  className="loader-wrapper"
>
  <div
    className="loader-wrapper__loader"
  >
    <div
      className="sk-spinner sk-folding-cube"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
    Loading...
  </div>
</div>
`;
